AzureCalling/Controllers/AudioDeviceSelectionViewController.swift

```diff
@@ -0,0 +1,131 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class AudioDeviceSelectionViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+
+    // MARK: Properties
+
+    private var audioDeviceOptions: [AudioDeviceOption] = [AudioDeviceOption]()
+    var deviceTable: UITableView!
+
+    // MARK: UIViewController events
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        self.modalTransitionStyle = .crossDissolve
+
+        view.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 0.8)
+        view.isOpaque = false
+
+        // tapping anywhere on the view is the same as tapping cancel
+        let tap = UITapGestureRecognizer(target: self, action: #selector(dismissSelf))
+        tap.cancelsTouchesInView = false
+        view.addGestureRecognizer(tap)
+        view.isUserInteractionEnabled = true
+
+        createAudioDeviceOptions()
+        createDeviceTable()
+    }
+
+    @objc func dismissSelf(sender: NSObject) {
+        dismiss(animated: true, completion: nil)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        openDeviceTable()
+    }
+
+    func createDeviceTable() {
+        deviceTable = UITableView()
+        deviceTable.isHidden = true
+        deviceTable.allowsSelection = true
+        deviceTable.isScrollEnabled = false
+        deviceTable.layer.cornerRadius = 8
+        deviceTable.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(deviceTable)
+        let audioDeviceCell = UINib(nibName: "BottomDrawerCellView",
+                                      bundle: nil)
+        deviceTable.register(audioDeviceCell, forCellReuseIdentifier: "BottomDrawerCellView")
+        deviceTable.dataSource = self
+        deviceTable.delegate = self
+        deviceTable.reloadData()
+
+        var deviceTableConstraints = [
+        deviceTable.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+        deviceTable.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+        deviceTable.heightAnchor.constraint(equalToConstant: 100)
+        ]
+
+        let hideConstraint = deviceTable.topAnchor.constraint(equalTo: view.bottomAnchor)
+        hideConstraint.priority = .defaultLow
+        deviceTableConstraints.append(hideConstraint)
+
+        NSLayoutConstraint.activate(deviceTableConstraints)
+    }
+
+    func openDeviceTable() {
+        deviceTable.isHidden = false
+
+        let centerYConstraint = NSLayoutConstraint(item: deviceTable!,
```

Comment:
`CenterYConstraint` means on the view centre. But here is not in centre, instead is bottom align. We could rename this constraint